libsrc/apple2/ser/a2.ssc.s

```diff
@@ -210,6 +193,48 @@ SER_CLOSE:
         stx     Index           ; Mark port as closed
         rts
 
+;----------------------------------------------------------------------------
+; Internal function to test whether we're really talking with a 6551,
+; or at least it really looks like it.
+; Input: X= Index
+; Output: carry set if this is very probably a 6551, clear if not
+
+Verify6551:
+        lda     ACIA_STATUS,x   ; Save current values in what we expect to be
+        pha                     ; the ACIA status register
+        lda     ACIA_CMD,x      ; and command register. So we can restore them
+        sta     tmp1            ; if this isn't a 6551.
+
+        and     #$01
+        bne     NotAcia         ; We expect command register bit 0 to be 0
```

Comment:
I don't think it's valid to assume that the 6551 is correctly reset when the driver gets loaded. I'd certainly start setting bit 0 before assuming any value.